docs/source/daml/intro/4_Transformations.rst

```diff
@@ -0,0 +1,154 @@
+
+.. literalinclude:: daml/4_Transformations/Contact.daml
+  :language: daml
+  :start-after: -- CHOICE_BEGIN
+  :end-before: -- CHOICE_END
+
+There's a lot to unpack in the above.
+
+- The first line, ``controller owner can`` says that the following choices are *controlled* by ``owner``, meaning ``owner`` is the party that may *exercise* them. The line starts a new block in which multiple choices can be defined.
+- ``UpdateTelephone`` is the name of a choice. It starts a new block in which that choice is defined.
+- ``: ContractId Contact`` is the return type of the choice. This choice archives the current ``Contact`` and creates a new one. It returns a reference to the new contract in the form of a ``ContractId Contact``
+- The following ``with`` block is that of a record. Just like with templates, in the background, a new record type is declared: ``data UpdateTelephone = UpdateTelephone with``
+- The `do` starts a block defining the action the choice should perform when exercised. In this case a new ``Contact`` is created.
+- The new ``Contact`` is created using ``this with``. ``this`` is a special value available within the ``where`` block of templates and takes the value of the current contract's arguments.
+
+There is nothing here indicating that the current ``Contact`` should be archived. That's because choices are *consuming* by default. That means when the above choice is exercised on a contract, that contract is archived.
+
+If you paid a lot of attention in :doc:`3_Data`, you may have noticed that the ``create`` statement returns an ``Update (ContractId Contact)``, not a ``ContractId Contact``. As a ``do`` block always returns the value of the last statement within it, the whole ``do`` block returns an ``Update``, but the return type on the choice is just a ``ContractId Contact``. This is a conveneience. Choices *always* return an ``Update`` so for readability it's omitted on the type declaration of a choice.
+
+Now to exercise the new choice in a scenario:
+
+.. literalinclude:: daml/4_Transformations/Contact.daml
+  :language: daml
+  :start-after: -- CHOICE_TEST_BEGIN
+  :end-before: -- CHOICE_TEST_END
+
+Choices are exercised using the ``exercise`` function, which takes a ``ContractId a``, and a value of type ``c``, where ``c`` is a choice on template ``a``. Since ``c`` is just a record, you can also just fill in the choice parameters using the ``with`` syntax you are already familiar with. ``exercise`` returns an ``Update r`` where ``r`` is the return type specified on the choice, allowing the new ``ContractId Contact`` to be stored in the variable ``new_acontactCid``.
```

Comment:
Active voice, breaks, typo
```suggestion
You exercise choices using the ``exercise`` function, which takes a ``ContractId a``, and a value of type ``c``, where ``c`` is a choice on template ``a``. Since ``c`` is just a record, you can also just fill in the choice parameters using the ``with`` syntax you are already familiar with.

``exercise`` returns an ``Update r`` where ``r`` is the return type specified on the choice, allowing the new ``ContractId Contact`` to be stored in the variable ``new_contactCid``.
```